### ISSUE TITLE
FxCopAnalyzers slowing down build and Visual Studio

### ISSUE THREAD

**Issue by JohanLarsson** [dotnet/roslyn-analyzers#3048]
With FxCopAnalzyers

![image](https://user-images.githubusercontent.com/1640096/68997690-83aefd00-08a9-11ea-9e3d-7a5573b15d06.png)

Without:

```xml
<!--<PackageReference Include="Microsoft.CodeAnalysis.FxCopAnalyzers" Version="2.9.7" PrivateAssets="all" />-->
```
![image](https://user-images.githubusercontent.com/1640096/68997697-96c1cd00-08a9-11ea-8f05-c73f6e6276c0.png)

Build time goes from 177 s to 1.8 s when I disable. Editor experience in Visual Studio is also greatly improved with the analyzer disabled, typing lag goes down significantly.

Attaching traces run with `FxCopAnalyzers` enabled and disabled. Used command `dotnet build --no-restore`

[WithFxCop.etl.zip](https://github.com/dotnet/roslyn-analyzers/files/3854576/WithFxCop.etl.zip)
[WithoutFxCop.etl.zip](https://github.com/dotnet/roslyn-analyzers/files/3854577/WithoutFxCop.etl.zip)

**Comment by JohanLarsson** [comment]
Ran with `/p:ReportAnalyzer=true`

![image](https://user-images.githubusercontent.com/1640096/68998516-5ebf8780-08b3-11ea-9978-266c0b5404c5.png)

**Comment by mavasani** [comment]
This will likely get addressed with https://github.com/dotnet/roslyn-analyzers/pull/3788. I will point you to a package to try out once that PR is merged. Thanks!

**Comment by mavasani** [comment]
Can you please try out https://dev.azure.com/dnceng/public/_packaging?_a=package&feed=dotnet5&package=Microsoft.CodeAnalysis.FxCopAnalyzers&protocolType=NuGet&version=3.3.0-beta1.20326.1&view=overview?

Also, if that doesn't help, can you check if adding `dotnet_code_quality.interprocedural_analysis_kind = None` to your .editorconfig resolves the issue? Thanks!

**Comment by JohanLarsson** [comment]
I no longer remember where the repro solution is. Nice that there was progress on this!
I'm thinking we close this issue and reopen or create a new issue if the problems show up again.
